kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/clusternetworkaddressconfigprovider/RangeAwarePortPerNodeClusterNetworkAddressConfigProvider.java

```diff
@@ -47,108 +49,65 @@
  *    <li>{@code nodeIdRanges} (required) defines the node id ranges present in the target cluster</li>
  * </ul>
  */
-public class RangeAwarePortPerNodeClusterNetworkAddressConfigProvider implements ClusterNetworkAddressConfigProvider {
-
-    private final HostPort bootstrapAddress;
-    private final String nodeAddressPattern;
-    private final Set<Integer> exclusivePorts;
-    private final Map<Integer, Integer> nodeIdToPort;
-
-    /**
-     * Creates the provider.
-     *
-     * @param config configuration
-     */
-    public RangeAwarePortPerNodeClusterNetworkAddressConfigProvider(RangeAwarePortPerNodeClusterNetworkAddressConfigProviderConfig config) {
-        this.bootstrapAddress = config.bootstrapAddress;
-        this.nodeAddressPattern = config.nodeAddressPattern;
-        this.nodeIdToPort = config.nodeIdToPort;
-        var allExclusivePorts = new HashSet<>(nodeIdToPort.values());
-        allExclusivePorts.add(bootstrapAddress.port());
-        this.exclusivePorts = Collections.unmodifiableSet(allExclusivePorts);
-    }
-
-    @Override
-    public HostPort getClusterBootstrapAddress() {
-        return this.bootstrapAddress;
-    }
-
-    @Override
-    public HostPort getBrokerAddress(int nodeId) throws IllegalArgumentException {
-        if (!nodeIdToPort.containsKey(nodeId)) {
-            throw new IllegalArgumentException(
-                    "Cannot generate node address for node id %d as it is not contained in the ranges defined for provider with downstream bootstrap %s"
-                            .formatted(
-                                    nodeId,
-                                    bootstrapAddress));
-        }
-        int port = nodeIdToPort.get(nodeId);
-        return new HostPort(BrokerAddressPatternUtils.replaceLiteralNodeId(nodeAddressPattern, nodeId), port);
-    }
-
-    @Override
-    public Set<Integer> getExclusivePorts() {
-        return this.exclusivePorts;
-    }
+@Plugin(configType = RangeAwarePortPerNodeClusterNetworkAddressConfigProvider.RangeAwarePortPerNodeClusterNetworkAddressConfigProviderConfig.class)
+public class RangeAwarePortPerNodeClusterNetworkAddressConfigProvider implements
```

Comment:
A bit clunky, I left the classname unmodified since we use the simple name to identify the plugin, this keeps the configuration YAML backwards compatible but means we have `*ClusterNetworkAddressConfigProvider` that implement `ClusterNetworkAddressConfigProviderService`.